### PR TITLE
First stab at Ready2Run conda package.

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

**Comment by cgranade** [comment]
Note, this PR will likely fail until microsoft/jupyter-core#37 is completed and until this kernel is updated to use the newest Jupyter.Core version, as KernelApplication currently calls an ASP.NET Core method that was removed with 3.0.

**Comment by cgranade** [comment]
With microsoft/jupyter-core#37 merged in, we should hopefully have everything we need for this build to work.

**Comment by cgranade** [comment]
NB: The pipeline for this PR has not yet been updated to enable building conda packages by default, so the status check is not yet accurate.

**Comment by cgranade** [comment]
This should now be ready, and the CI pipeline should be correctly reporting its status. @bettinaheim and @anpaz-msft, could you please take another look? Thanks! 💕

File: build/pack.ps1 [hunk 1]
```diff
@@ -79,12 +79,40 @@ function Pack-Image() {
...
+        [string] $Configuration = $Env:BUILD_CONFIGURATION
+    );
+
+    $OutputPath = Join-Path $Env:BLOBS_OUTDIR $Runtime;
+
+    # Suppress generating pdb files.
+    # See https://github.com/dotnet/cli/issues/2246#issuecomment-320633639.
+    dotnet publish `
+    (Join-Path $PSScriptRoot $Project) `
+        -c $Configuration `
```
Comment (anpaz-msft) [inline build/pack.ps1]: I'm pretty sure you're missing passing the assembly version.
I think you will also need to pass assembly constants for signed/telemetry.

Comment (anpaz-msft) [reply build/pack.ps1]: these are all the flags we normally use:
```
        -v $Env:BUILD_VERBOSITY `
        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
        /property:Version=$Env:ASSEMBLY_VERSION `
```

File: build/steps-conda.yml [hunk 2]
```diff
@@ -7,24 +7,35 @@ steps:
 ##
 # Pre-reqs
 ##
+- task: DownloadBuildArtifacts@0
+  inputs:
+    artifactName: iqsharp
+    downloadPath: '$(System.DefaultWorkingDirectory)/artifact'
+
+- pwsh: |
```
Comment (anpaz-msft) [inline build/steps-conda.yml]: how is this going to work with the end-to-end/signed build? name of the drops is different. 
Also, should this be inside a script so we dont' have to copy the same commands?

File: build/pack.ps1 (hunk 1 above)
Comment (cgranade) [reply build/pack.ps1]: Thanks for the catch, somehow I had thought those propagated from the build to the publish step, but that would explain it. I'll fix that now, thanks!

File: build/steps-conda.yml (hunk 2 above)
Comment (cgranade) [reply build/steps-conda.yml]: With respect to copying the same commands, my thinking was that it's a single-line script whose parameters vary between the signed and prerelease builds as per your comment, such that making a new script would not save very much over copying and pasting. Happy to discuss that further, though.

**Pull request merged**